### ISSUE TITLE
[Bug]: Shared folder not listing available shraes

### ISSUE THREAD

**Issue by cybertschunk** [nextcloud/server#35486]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

We just updated to Nextcloud 24.0.7. we are running the Community Docker Image.
Our users authenticate against a LDAP authority. Unfortunately since one of the last updates the sharing feature is broken. The shared folder is displayed as empty in the webinterface.
![Screenshot 2022-11-29 at 13-39-10 Shared - Dateien - Nextcloud](https://user-images.githubusercontent.com/21140322/204531653-9b287591-c6fb-4aa7-b52d-f2e63959c46b.png)
 In the mobile app however, the available shares are still displayed. Same applies to the "Shared with you option 
![Screenshot 2022-11-29 at 13-38-32 Mit dir geteilt - Nextcloud](https://user-images.githubusercontent.com/21140322/204531755-95c2e1df-a834-4c1d-837c-a13d94d2babc.png)
If the user tries to click on the displayed folder, the operation fails. He just sees the empty "Shared" folder.

### Steps to reproduce

1. Share a folder with another user
2. log in as the other user
3. Open the Shared folder -> empty
4. Open the "Shared with you" menu entry and click on the shared folder -> nothing is displayed
5.  download the folder in the "shared with you" menu -> everything works 

### Expected behavior

The user can see his shares in the Shared folder and open them

### Installation method

Community Docker image

### Operating system

Debian/Ubuntu

### Web server

Nginx

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 22.2.3 to 22.2.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "24.0.7.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "ldapIgnoreNamingRules": false,
        "loglevel": 0,
        "theme": "",
        "defaultapp": "files",
        "maintenance": false,
        "trusted_domains": [
            "files.asta.uni-luebeck.de",
            "141.83.153.154",
            "app-server",
            "nginx-serer",
            "onlyoffice-document-server"
        ],
        "allow_local_remote_servers": true,
        "share_folder": "\/Shared",
        "mail_smtpmode": "smtp",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "forcessl": true,
        "trashbin_retention_obligation": "auto",
        "logtimezone": "Europe\/Berlin",
        "enable_previews": true,
        "updatechecker": false,
        "singleuser": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "overwritehost": "files.asta.uni-luebeck.de",
        "overwriteprotocol": "https",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "tls",
        "mail_smtpport": "587",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "debug": false,
        "onlyoffice": {
            "verify_peer_off": true
        },
        "csrf.optout": [
            "\/^WebDAVFS\/",
            "\/^Microsoft-WebDAV-MiniRedir\/",
            "\/KeeWeb\\\/\\d+[.]\\d+[.]\\d+\/"
        ],
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "overwrite.cli.url": "https:\/\/files.asta.uni-luebeck.de",
        "mail_sendmailmode": "smtp"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - appointments: 1.14.3
  - bruteforcesettings: 2.4.0
  - calendar: 3.5.2
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - dav: 1.22.0
  - deck: 1.7.3
  - external: 4.0.1
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_external: 1.16.1
  - files_markdown: 2.3.6
  - files_pdfviewer: 2.5.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - firstrunwizard: 2.13.0
  - forms: 2.5.1
  - groupfolders: 12.0.2
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - mail: 1.14.3
  - notifications: 2.12.1
  - oauth2: 1.12.0
  - password_policy: 1.14.0
  - photos: 1.6.0
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - recommendations: 1.3.0
  - richdocuments: 6.3.1
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - support: 1.7.0
  - systemtags: 1.14.0
  - text: 3.5.1
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - twofactor_totp: 6.4.1
  - updatenotification: 1.14.0
  - user_ldap: 1.14.1
  - user_status: 1.4.0
  - viewer: 1.8.0
  - weather_status: 1.4.0
  - workflowengine: 2.6.0
Disabled:
  - admin_audit
  - circles: 24.0.1
  - encryption
  - nextcloud_announcements: 1.5.0
  - survey_client: 1.4.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
{"reqId":"tLU71Zt0KzP5KZFbNBp3","level":0,"time":"2022-11-29T13:47:44+01:00","remoteAddr":"REMOVeD","user":"REMOVED","app":"user_ldap","method":"GET","url":"/ocs/v2.php/apps/notifications/api/v2/notifications","message":"Ready for a paged search","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:107.0) Gecko/20100101 Firefox/107.0","version":"24.0.7.1","data":{"app":"user_ldap"}}
{"reqId":"tLU71Zt0KzP5KZFbNBp3","level":0,"time":"2022-11-29T13:47:44+01:00","remoteAddr":"141.83.38.137","user":"7ef84856-b087-1039-9f09-c98518c5d872","app":"user_ldap","method":"GET","url":"/ocs/v2.php/apps/notifications/api/v2/notifications","message":"readAttribute: REMOVED found","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:107.0) Gecko/20100101 Firefox/107.0","version":"24.0.7.1","data":{"app":"user_ldap"}}
{"reqId":"oLYfaL6AdzJAqrhwbNyU","level":3,"time":"2022-11-29T13:47:53+01:00","remoteAddr":"REMOVED","user":"REMOVED","app":"files_sharing","method":"PROPFIND","url":"/remote.php/dav/files/6c07b1b8-61b5-1038-80f6-796b4f1a017e/","message":"Invalid recipient","userAgent":"Mozilla/5.0 (Windows) mirall/2.6.3stable-Win64 (build 20200217) (Nextcloud)","version":"24.0.7.1","exception":{"Exception":"InvalidArgumentException","Message":"Invalid recipient","Code":0,"Trace":[{"file":"/var/www/html/apps/files_sharing/lib/SharedMount.php","line":156,"function":"moveShare","class":"OC\\Share20\\Manager","type":"->","args":[{"__class__":"OC\\Share20\\Share"},"6c07b1b8-61b5-1038-80f6-796b4f1a017e"]},{"file":"/var/www/html/apps/files_sharing/lib/SharedMount.php","line":136,"function":"updateFileTarget","class":"OCA\\Files_Sharing\\SharedMount","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/apps/files_sharing/lib/SharedMount.php","line":89,"function":"verifyMountPoint","class":"OCA\\Files_Sharing\\SharedMount","type":"->","args":["*** sensitive parameters replaced ***",[{"__class__":"OCA\\Files_Sharing\\SharedMount"},{"__class__":"OCA\\Files_Sharing\\SharedMount"},{"__class__":"OCA\\Files_Sharing\\SharedMount"},{"__class__":"OCA\\Files_Sharing\\SharedMount"},{"__class__":"OCA\\Files_Sharing\\SharedMount"}],{"__class__":"OC\\Cache\\CappedMemoryCache"}]},{"file":"/var/www/html/apps/files_sharing/lib/MountProvider.php","line":147,"function":"__construct","class":"OCA\\Files_Sharing\\SharedMount","type":"->","args":["\\OCA\\Files_Sharing\\SharedStorage",[{"__class__":"OCA\\Files_Sharing\\SharedMount"},{"__class__":"OCA\\Files_Sharing\\SharedMount"},{"__class__":"OCA\\Files_Sharing\\SharedMount"},{"__class__":"OCA\\Files_Sharing\\SharedMount"},{"__class__":"OCA\\Files_Sharing\\SharedMount"}],["6c07b1b8-61b5-1038-80f6-796b4f1a017e","*** sensitive parameters replaced ***",[{"__class__":"OC\\Share20\\Share"}],{"__class__":"OC\\Files\\View"},false],{"__class__":"OC\\Files\\Storage\\StorageFactory"},{"__class__":"OC\\Files\\View"},{"__class__":"OC\\Cache\\CappedMemoryCache"},{"__class__":"OC\\EventDispatcher\\EventDispatcher"},{"__class__":"OC\\User\\User"},{"__class__":"OC\\Memcache\\APCu"}]},{"file":"/var/www/html/lib/private/Files/Config/MountProviderCollection.php","line":134,"function":"getMountsForUser","class":"OCA\\Files_Sharing\\MountProvider","type":"->","args":[{"__class__":"OC\\User\\User"},{"__class__":"OC\\Files\\Storage\\StorageFactory"}]},{"file":"/var/www/html/lib/private/Files/SetupManager.php","line":225,"function":"addMountForUser","class":"OC\\Files\\Config\\MountProviderCollection","type":"->","args":[{"__class__":"OC\\User\\User"},{"__class__":"OC\\Files\\Mount\\Manager"},{"__class__":"Closure"}]},{"file":"/var/www/html/lib/private/Files/SetupManager.php","line":317,"function":"OC\\Files\\{closure}","class":"OC\\Files\\SetupManager","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/lib/private/Files/SetupManager.php","line":226,"function":"setupForUserWith","class":"OC\\Files\\SetupManager","type":"->","args":[{"__class__":"OC\\User\\User"},{"__class__":"Closure"}]},{"file":"/var/www/html/lib/private/legacy/OC_Util.php","line":111,"function":"setupForUser","class":"OC\\Files\\SetupManager","type":"->","args":[{"__class__":"OC\\User\\User"}]},{"file":"/var/www/html/apps/dav/lib/Connector/Sabre/Auth.php","line":123,"function":"setupFS","class":"OC_Util","type":"::","args":["6c07b1b8-61b5-1038-80f6-796b4f1a017e"]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Auth/Backend/AbstractBasic.php","line":103,"function":"validateUserPass","class":"OCA\\DAV\\Connector\\Sabre\\Auth","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/html/apps/dav/lib/Connector/Sabre/Auth.php","line":251,"function":"check","class":"Sabre\\DAV\\Auth\\Backend\\AbstractBasic","type":"->","args":[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"/var/www/html/apps/dav/lib/Connector/Sabre/Auth.php","line":154,"function":"auth","class":"OCA\\DAV\\Connector\\Sabre\\Auth","type":"->","args":[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Auth/Plugin.php","line":180,"function":"check","class":"OCA\\DAV\\Connector\\Sabre\\Auth","type":"->","args":[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Auth/Plugin.php","line":135,"function":"check","class":"Sabre\\DAV\\Auth\\Plugin","type":"->","args":[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"/var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"beforeMethod","class":"Sabre\\DAV\\Auth\\Plugin","type":"->","args":[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php","line":456,"function":"emit","class":"Sabre\\DAV\\Server","type":"->","args":["beforeMethod:PROPFIND",[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->","args":[{"__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"/var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/apps/dav/lib/Server.php","line":358,"function":"exec","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/remote.php","line":170,"args":["/var/www/html/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/html/lib/private/Share20/Manager.php","Line":1297,"CustomMessage":"--"}}
{"reqId":"oLYfaL6AdzJAqrhwbNyU","level":4,"time":"2022-11-29T13:47:53+01:00","remoteAddr":"141.83.153.131","user":"REMOVED","app":"files_sharing","method":"PROPFIND","url":"/remote.php/dav/files/REMOVED/","message":"Error while trying to create shared mount","userAgent":"Mozilla/5.0 (Windows) mirall/2.6.3stable-Win64 (build 20200217) (Nextcloud)","version":"24.0.7.1","data":{"app":"files_sharing"}}
```

**Comment by szaimen** [comment]
Will be fixed with 24.0.8

**Comment by szaimen** [comment]
And yes everything was correctly filled in, thansk! :)
